### DISCUSSION TITLE
Jumpstarting a @genome-spy/app with groups and server-side bookmarks

### DISCUSSION THREAD

**Discussion by razsultana** [genome-spy/genome-spy#253] [Q&A] [answered]
I am using the _amazing_ GenomeSpy app to create a visualisation that includes several tracks, which I would like to be grouped  at two levels, based on the cell line that was used and whether the experiment that generated the data was a control or a test. 
While it's easy to do this in the user interface, by right-clicking the right metadata columns and using group on them, I would like this to be the initial visualisation status, but I can't find any mention of how to accomplish that in the documentation. 

I suspect  I could do it through the `options` parameter supplied to the embed function when the app is created:
https://github.com/genome-spy/genome-spy/blob/cd145b5f46c471ff79d10529f1c709124e0d4c86/packages/app/src/index.js#L19
but I'm not quite sure how those options should look like.

The second part of this question is about storing bookmarks on the server side vs the user's browser local storage. 
I can't find in the documentation how I can accomplish that, either interactively or programatically (through some options/config files). Any hints about how to do that would be greatly appreciated.
If there was a way to save the bookmarks on the server, that would almost solve my first problem as well, as the server-side bookmark could include groupings, filtering, etc.

**Comment by razsultana** [comment, marked as answer]
I actually found the answer to my second question in the documentation:
https://genomespy.app/docs/sample-collections/visualizing/#pre-defined-bookmarks-and-bookmark-tour
It seems pretty straightforward and it also seems to solve my first problem as a default server-side bookmark can be loaded on the beginning of the visualisation.

**Comment by tuner** [comment]
Yeh. The bookmark system should be quite flexible. Let me know if you think the documentation needs any clarification.

**Reply by razsultana** [reply]
Yes, the bookmarks system solved my problem of setting an initial state beautifully. 
I just set up in the spec of the visualisation:
```
 "bookmarks": {
    "remote": {
      "url": "bookmarks.json",
      "initialBookmark": "Initial state",
      "tour": false
    }
  },
```
Then I set up the groupings and zoom level interactively, clicked the "share" button, then selected the JSON tab and copied the bookmark content from there, then saved it in my bookmarks.json as such.
I think the documentation is actually quite good, I'm not sure why I didn't see it in the first place. 
The only thing that was not so apparent was the `initialBookmark` property - it's mentioned but not emphasised.

I think the use case of setting a "standard" state when starting the visualisation is going to be very common, so that might be something to add to the documentation as a separate example from just "server-side bookmarks" case.
